src/config.c

```diff
@@ -1821,18 +1821,25 @@ static int sdsConfigSet(standardConfig *config, sds *argv, int argc, const char
     UNUSED(argc);
     if (config->data.sds.is_valid_fn && !config->data.sds.is_valid_fn(argv[0], err))
         return 0;
-    sds prev = config->flags & MODULE_CONFIG ? getModuleStringConfig(config->privdata) : *config->data.sds.config;
+    sds old = config->flags & MODULE_CONFIG ? getModuleStringConfig(config->privdata) : *config->data.sds.config;
     sds new = (config->data.string.convert_empty_to_null && (sdslen(argv[0]) == 0)) ? NULL : argv[0];
-    if (new != prev && (new == NULL || prev == NULL || sdscmp(prev, new))) {
-        sdsfree(prev);
-        if (config->flags & MODULE_CONFIG) {
-            return setModuleStringConfig(config->privdata, new, err);
-        }
-        *config->data.sds.config = new != NULL ? sdsdup(new) : NULL;
-        return 1;
+
+    /* Return if old and new config are equals */
+    if ((old == new) || (new != NULL && old != NULL && (sdscmp(old, new) == 0))) {
```

Comment:
i think it could be confusing that this setter is now different in layout from other setters (e.g. `stringConfigSet`).
someone modifying them all, might make a mistake.
i'd rather revert the layout, and just add comments about the odd concerns about freeing the previous or temp pointers.

Reply:
Ok. I will revert and just the comment.